### ISSUE TITLE
ls doesn't follow links to directories

### ISSUE THREAD

**Issue by nfischer** [shelljs/shelljs#733]
**Originally reported as #446**

### Node version (or tell us if you're using electron or some other framework):

latest

### ShellJS version (the most recent version/Github branch you see the bug on):

Latest. The bug was introduced in v0.7.0

### Operating system:

Confirmed on both OS X and Linux.

### Description of the bug:

Originally reported in #446. I'm branching this off, since that issue actually discusses two changes.

`ls()` doesn't follow symlinks pointing to directories, while `ls` does in Bash.

### Example ShellJS command to reproduce the error:

```javascript
// setup
shell.mkdir('real');
shell.touch('real/file1');
shell.touch('real/file2');
shell.ln('-s', 'real', 'sym');

shell.ls('real');      // file1, file2
shell.ls('sym');       // sym (this is a bug)
shell.exec('ls real'); // file1, file2 (bash behavior)
shell.exec('ls sym');  // file1, file2 (bash behavior)
```

@freitagbr can you investigate? We should bisect to find the commit that introduced this bug (possibly something from [this list](https://github.com/shelljs/shelljs/issues?utf8=%E2%9C%93&q=is%3Aclosed%20link%20ls%20)). From there, we can figure out how to fix this.

**Comment by freitagbr** [comment]
It has to do with the way files are handled, specificallly in `pushFile`. `fs.lstatSync` is used in the default case, which returns stats about the symbolic link itself, not the file or directory it links to. I think there needs to be extra handling for the case when `stat.isSymbolicLink()` is `true`. I will work on a fix.

**Comment by nfischer** [comment]
@freitagbr were you able to bisect to a commit? Unix behavior can be somewhat inconsistent around here, so this bug probably arose as a side effect of another bug fix. It would be good to understand all the scenarios we need to account for.

**Comment by freitagbr** [comment]
I will do the bisect tonight.

**Comment by freitagbr** [comment]
Ok, git bisect landed me at commit `3e441e4`. I used a test script to help out:
```
$ mkdir real
$ touch real/a
$ touch real/b
$ ln -s real sym
$ cat test-ls.js
var assert = require('assert');
var $ = require('./');
var result = $.ls('./sym');
assert(result.indexOf('a') > -1);
assert(result.indexOf('b') > -1);
$ git bisect master v0.5.0  # v0.5.0 is a known good revision
$ git bisect run node test-ls.js
```

**Comment by nfischer** [comment]
That would be #369
